src/named_tuple.cr

```diff
@@ -38,13 +38,15 @@ struct NamedTuple
   end
 
   # Creates a named tuple from the given hash, with elements casted to the given types.
+  # Here the Int32 | String union is cast to Int32.
   #
   # ```
-  # NamedTuple(foo: String, bar: Int64).from({:foo => "world", :bar => 2})       # => {foo: "world", bar: 2}
-  # NamedTuple(foo: String, bar: Int64).from({"foo" => "world", "bar" => 2})     # => {foo: "world", bar: 2}
-  # NamedTuple(foo: String, bar: Int64).from({:foo => "world", :bar => 2}).class # => {foo: String, bar: Int64}
-  # ```
+  # num_or_str = 42.as(Int32 | String)
+  # NamedTuple(name: String, val: Int32).from({"name" => "number", "val" => num_or_str}) # => {name: "number", val: 42}
```

Comment:
New examples are using `Hash` instead of `NamedTuple` for an argument. Also, I'd stick to one-liners in this particular case, otherwise you must to reread previous line to get the value of `num_or_str` instead of just having it inline along with all expression.